lib/signaling/v2/peerconnection.js

```diff
@@ -955,7 +954,20 @@ class PeerConnectionV2 extends StateMachine {
           return Promise.resolve();
         }
         if (this._peerConnection.signalingState === 'have-local-offer') {
-          if (this._needsAnswer && this._descriptionRevision === 1) {
+          // For a peer connection -
+          // 1) createOffer always generate SDP with `setup:actpass`
+          // 2) when remote description is set `setup:active`  - the answer generated selects the dtls role of setup:passive
+          // 3) when remote description is set `setup:passive` - the answer generated selects the dtls role of setup:active
+          // 4) when remote description is set `setup:actpass` - the answer generated uses the previously negotiated role (if not negotiated previously setup:active is used)
+          // This test makes the behavior super clear.
+          // https://github.com/twilio/twilio-webrtc.js/blob/master/test/integration/spec/rtcpeerconnection.js#L936
+          // if (this._needsAnswer && this._descriptionRevision === 1) {
+          //   this._queuedDescription = description;
+          //   return Promise.resolve();
+          // }
+          // with glare handling (if dtls role was not negotiated before ) the generated answer will set setup:active.
+          // we do not want that. lets wait for "initial negotiation" before attempting glare handling.
+          if (this._needsAnswer && this._lastStableDescriptionRevision === 0) {
```

Comment:
Question: Would `this._lastStableDescriptionRevision` be 0 after VMS Failover if at least 1 SDP negotiation already happened? I do not see it being reset to 0 anywhere in the code. More unit/integration tests would help find the answer, I guess

Reply:
Yes. For new PC, we set this 0.  https://github.com/twilio/twilio-video.js/blob/f300dc0e161b09d4e77d85405b2c95e7834f529c/lib/signaling/v2/peerconnection.js#L207, 

it will be updated only after new PC completes a negotiation.

Reply:
Good point. @innerverse - is revision 0 in RSP a possibility?

Reply:
We always start `create-offer` at rev1.  You should be fine with this.